src/backend/access/index/indexam.c

```diff
@@ -644,11 +645,20 @@ index_getbitmap(IndexScanDesc scan, Node *bitmap)
 	/*
 	 * have the am's getbitmap proc do all the work.
 	 */
-	bm = (Node *) DatumGetPointer(FunctionCall2(procedure,
-									  PointerGetDatum(scan),
-									  PointerGetDatum(bitmap)));
+	d = FunctionCall2(procedure,
+					  PointerGetDatum(scan),
+					  PointerGetDatum(bitmapP));
 
-	return bm;
+	ntids = DatumGetInt64(d);
+
+	/* If int8 is pass-by-ref, must free the result to avoid memory leak */
+#ifndef USE_FLOAT8_BYVAL
+	pfree(DatumGetPointer(d));
+#endif
```

Comment:
This part [came](https://github.com/postgres/postgres/blob/REL9_4_STABLE/src/backend/access/index/indexam.c#L658) from vanilla Postgres. `Datum` [type](https://github.com/arenadata/gpdb/blob/b33ee7afa13b0aa813245a0a3c783aa3cbb00d01/src/include/postgres.h#L329) can be used for both, passing int variable by reference or by value. As a part of optimization, Postgres has `USE_FLOAT8_BYVAL` flag preprocessing, which, if defined, allows to pass int variables by value. It means, when it's not defined, we need to free a pointer to avoid memory leak.

There is an interesting [comment](https://github.com/arenadata/gpdb/blob/b33ee7afa13b0aa813245a0a3c783aa3cbb00d01/src/include/pg_config.h.in#L931), which says this flag is always defined for GPDB. Despite of that, GPDB codebase is full of `USE_FLOAT8_BYVAL` flag preprocessing, so I think, it's better to keep analyzing it - to simpler merging or to have an ability to disable it one day.